src/Compiler/FSharp.Compiler.Service.fsproj

```diff
@@ -13,6 +13,9 @@
     <AssemblyName>FSharp.Compiler.Service</AssemblyName>
     <AllowCrossTargeting>true</AllowCrossTargeting>
     <DefineConstants>$(DefineConstants);COMPILER</DefineConstants>
+    <CheckNulls>true</CheckNulls>
+	<!-- Nullness checking againts ns20 base class libraries is very weak, the APIs were not updated with annotations. -->
+    <TargetFrameworks Condition=" '$(BuildingInsideVisualStudio)' == 'true' Or '$(DesignTimeBuild)' == 'true' ">net8.0;$(TargetFrameworks)</TargetFrameworks>
```

Comment:
Defer merging because of this.
The net8 APIs are needed to get good annotations, but we do not want/need to build the actual product with net8.

The condition above means it would not run as part of CI, which was done by me to not make the CI even slower.
However, this would also mean that errors visible only in net8 could sneak into main - not a good idea.